org.eclipse.ui.console/src/org/eclipse/ui/console/TextConsole.java

```diff
@@ -545,19 +539,39 @@ private synchronized void checkFinished() {
 	 * Adds a hyperlink to this console.
 	 *
 	 * @param hyperlink the hyperlink to add
-	 * @param offset the offset in the console document at which the hyperlink should be added
-	 * @param length the length of the text which should be hyperlinked
+	 * @param offset    the offset in the console document at which the hyperlink
+	 *                  should be added
+	 * @param length    the length of the text which should be hyperlinked
 	 * @throws BadLocationException if the specified location is not valid.
 	 */
 	public void addHyperlink(IHyperlink hyperlink, int offset, int length) throws BadLocationException {
-		IDocument document = getDocument();
 		ConsoleHyperlinkPosition hyperlinkPosition = new ConsoleHyperlinkPosition(hyperlink, offset, length);
-		try {
-			document.addPosition(ConsoleHyperlinkPosition.HYPER_LINK_CATEGORY, hyperlinkPosition);
-			fConsoleManager.refresh(this);
-		} catch (BadPositionCategoryException e) {
-			ConsolePlugin.log(e);
+		fPositions.add(getInsertionIndex(offset), hyperlinkPosition);
+		fConsoleManager.refresh(this);
+	}
+
+	private int getInsertionIndex(int offset) {
+		if (fPositions.isEmpty()) {
+			return 0;
 		}
+
+		var left = 0;
```

Comment:
same here

Reply:
yes in this case it is possible !

Reply:
in fact it is not possible, the binarySearch returns -1 if the key is not found. An other option would be to let ConsoleHyperlinkPosition implement Comparable and store positions in a set.

Reply:
done too !